examples/list_disk_snapshots.py

```diff
@@ -76,6 +80,21 @@ def parse_args():
             "status": str(s.status),
         }
         snapshots[parent] = child
+    
+    # Go through the other snapshots to detect possible items which are parents
+    # of snapshots added to the mapping, but have other disk.id (e.g. template
+    # disks)
+    for s in other_snapshots:
+        if s.id in snapshots:
+            parent = s.parent.id if s.parent else None
+            child = {
+                "actual_size": s.actual_size,
+                "format": str(s.format),
+                "id": s.id,
+                "parent": parent,
+                "status": str(s.status),
+            }
+            snapshots[parent] = child
```

Comment:
If other_snapshots is a dict, we can lookup the snapshot using the parent id stored in the snapshots dict.